tests/cephadm/test_redeploy_service.py

```diff
@@ -0,0 +1,54 @@
+import json
+
+from ceph.waiter import WaitUntil
+from cli.cephadm.cephadm import CephAdm
+
+
+class RedeployServiceError(Exception):
+    pass
+
+
+def verify_redeploy_demon(installer_node, redeploy_daemon):
+    """
+    Verify redeploy demon.
+    Args:
+        installer_node: installer node
+        redeploy_daemon: reploy demon
+    """
+    timeout, interval = 60, 5
+    for w in WaitUntil(timeout=timeout, interval=interval):
+        result = CephAdm(installer_node).ceph.orch.ps(format="json")
+        all_running_daemons = [
+            data["daemon_name"]
+            for data in json.loads(result[installer_node[0].hostname][0])
+        ]
+        if (item == redeploy_daemon for item in all_running_daemons):
+            break
+    if w.expired:
+        raise RedeployServiceError("Redeploy deamons check failed")
+
+
+def run(ceph_cluster, **kw):
+    """Verify redeploy for a specific service
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    installer_node = ceph_cluster.get_nodes(role="installer")
+    config = kw.get("config")
+    service_name = config.get("service_name")
```

Comment:
Add validation for parameters recieved from config

Reply:
Updated